policy-engine/src/graph.rs

```diff
@@ -61,29 +63,15 @@ pub(crate) fn index(req: HttpRequest) -> Box<Future<Item = HttpResponse, Error =
     if let Err(e) = commons::ensure_query_params(mandatory_params, req.query_string()) {
         return Box::new(future::err(e));
     }
-    // TODO(steveeJ): take another look at the actix-web docs for a method that
-    // provides this parameters split.
-    let plugin_params = req
-        .query_string()
-        .to_owned()
-        .split('&')
-        .map(|pair| {
-            let kv_split: Vec<&str> = pair.split('=').collect();
-
-            let value = kv_split
-                .get(1)
-                .unwrap_or_else(|| {
-                    trace!(
-                        "query parameter '{}' is not a k=v pair. assuming an empty value.",
-                        pair
-                    );
-                    &""
-                })
-                .to_string();
-
-            (kv_split[0].to_string(), value)
-        })
-        .collect();
+
+    let plugin_params = match Query::from_query(req.query_string()) {
+        Ok(query) => (query as Query<HashMap<String, String>>).into_inner(),
```

Comment:
Just a minor question: is there a way to move this type hint to the call site? Something like `Query<HashMap<String, String>>::from_query()` or a similar syntax, if that exists.

Reply:
This was indeed possible, thanks for the suggestion!